contracts/protocols/mainnet/morpho/helpers.sol

```diff
@@ -0,0 +1,369 @@
+        return 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE;
+    }
+
+    function getAaveLens() internal pure returns (address) {
+        return 0x507fA343d0A90786d86C7cd885f5C49263A91FF4;
+    }
+
+    function getCompoundLens() internal pure returns (address) {
+        return 0x930f1b46e1D081Ec1524efD95752bE3eCe51EF67;
+    }
+
+    function getCompMorpho() internal pure returns (address) {
+        return 0x8888882f8f843896699869179fB6E4f7e3B58888;
+    }
+
+    function getAaveMorpho() internal pure returns (address) {
+        return 0x777777c9898D384F785Ee44Acfe945efDFf5f3E0;
+    }
+
+    function getAaveProtocolDataProvider() internal pure returns (address) {
+        return 0x057835Ad21a177dbdd3090bB1CAE03EaCF78Fc6d;
+    }
+
+    function getAaveIncentivesController() internal pure returns (address) {
+        return 0xd784927Ff2f95ba542BfC824c8a8a98F3495f6b5;
+    }
+
+    function getComptroller() internal pure returns (address) {
+        return 0x3d9819210A31b4961b30EF54bE2aeD79B9c9Cd3B;
+    }
+
+    enum Underlying {
+        AAVEV2,
+        COMPOUNDV2
+    }
+
+    struct MorphoData {
+        MarketDetail[] aaveMarketsCreated;
+        MarketDetail[] compMarketsCreated;
+        bool isClaimRewardsPausedAave;
+        bool isClaimRewardsPausedComp;
+    }
+
+    struct TokenConfig {
+        address poolTokenAddress;
+        address underlyingToken;
+        uint256 decimals;
+    }
+
+    struct AaveMarketDetail {
+        uint256 aEmissionPerSecond;
+        uint256 vEmissionPerSecond;
+        uint256 availableLiquidity;
+        uint256 liquidityRate;
+        uint256 ltv;
+        uint256 liquidationThreshold;
+        uint256 liquidationBonus;
+    }
+
+    struct CompoundMarketDetail {
+        uint256 compSpeed;
+        uint256 compSupplySpeed;
+        uint256 compBorrowSpeed;
+        uint256 collateralFactor;
+    }
+
+    struct MarketDetail {
+        TokenConfig config;
+        uint256 avgSupplyRate; //in wad
+        uint256 avgBorrowRate; //in wad
+        uint256 p2pSupplyRate;
+        uint256 p2pBorrowRate;
+        uint256 poolSupplyRate;
+        uint256 poolBorrowRate;
+        uint256 totalP2PSupply;
+        uint256 totalPoolSupply;
+        uint256 totalP2PBorrows;
+        uint256 totalPoolBorrows;
+        uint256 poolSupplyIndex; //exchange rate of cTokens for compound
+        uint256 poolBorrowIndex;
+        uint256 p2pSupplyDelta; //The total amount of underlying ERC20 tokens supplied through Morpho,
+        //stored as matched peer-to-peer but supplied on the underlying pool
+        uint256 p2pBorrowDelta; //The total amount of underlying ERC20 tokens borrow through Morpho,
+        //stored as matched peer-to-peer but borrowed from the underlying pool
+        uint256 reserveFactor;
+        AaveMarketDetail aaveData;
+        CompoundMarketDetail compData;
+        Flags flags;
+    }
+
+    struct Flags {
+        bool isPaused;
+        bool isPartiallyPaused;
```

Comment:
@Richa-iitr what does partially paused mean here?

Reply:
@thrilok209 Partially paused is when only `repay` and `withdraw` is allowed and not other intercations.

Reply:
Got it!